File: VB/WinSolution.Module/Updater.vb
Imports Microsoft.VisualBasic
Imports System
Imports DevExpress.Xpo
Imports DevExpress.ExpressApp.Updating

Namespace WinSolution.Module
	Public Class Updater
		Inherits ModuleUpdater
		Public Sub New(ByVal session As Session, ByVal currentDBVersion As Version)
			MyBase.New(session, currentDBVersion)
		End Sub
		Public Overrides Sub UpdateDatabaseAfterUpdateSchema()
			MyBase.UpdateDatabaseAfterUpdateSchema()
			InitData()
		End Sub
		Private Sub InitData()
			Using uow As New UnitOfWork(Session.DataLayer)
				For i As Integer = 0 To 99999
					Dim product As New Product(uow)
					product.Name = String.Format("Product{0}", i)
					product.Description = String.Format("Description{0}", i)
					product.Price = i Mod 10
				Next i
				uow.CommitChanges()
			End Using
		End Sub
	End Class
End Namespace

File: CS/WinSolution.Module/Updater.cs
using System;
using DevExpress.Xpo;
using DevExpress.ExpressApp.Updating;

namespace WinSolution.Module {
    public class Updater : ModuleUpdater {
        public Updater(Session session, Version currentDBVersion) : base(session, currentDBVersion) { }
        public override void UpdateDatabaseAfterUpdateSchema() {
            base.UpdateDatabaseAfterUpdateSchema();
            InitData();
        }
        private void InitData() {
            using (UnitOfWork uow = new UnitOfWork(Session.DataLayer)) {
                for (int i = 0; i < 100000; i++) {
                    Product product = new Product(uow);
                    product.Name = string.Format("Product{0}", i);
                    product.Description = string.Format("Description{0}", i);
                    product.Price = i % 10;
                }
                uow.CommitChanges();
            }
        }
    }
}